File: Demo/Sources/ViewControllers/DataView/List/DataProvider/ListDataProvider.m
//
//  Globus
//

#import "ListDataProvider.h"

@implementation ListDataProvider

+ (instancetype)dataProvider {
    ListDataProvider* dataProvider = [self new];
    dataProvider.canReload = YES;
    dataProvider.canSearch = YES;
    return dataProvider;
}

- (id)modelWithJsonObject:(id)jsonObject {
    return [ListDataProviderModel modelWithJson:jsonObject];
}

@end

@implementation ListDataProviderModel

#pragma mark - GLBModel

+ (NSDictionary< NSString*, GLBModelJson* >*)jsonMap {
    return @{
        @"uid" : [GLBModelJsonString jsonWithPath:@"id"],
        @"title" : [GLBModelJsonString jsonWithPath:@"title"]
    };
}

@end

File: Demo/Sources/ViewControllers/DataView/List/DataProvider/ListDataProvider.swift
//
//  Globus
//

import Globus
import GlobusSwifty

class ListDataProvider: GLBLocalListDataProvider {
    
    public override init() {
        super.init(baseFileName: "ListDataProvider")
        self.canReload = true
        self.canSearch = true
    }
    
    public override init(baseFileName: String) {
        super.init(baseFileName: baseFileName)
        self.canReload = true
        self.canSearch = true
    }
    
    public override func model(withJsonObject jsonObject: Any) -> Any? {
        return ListDataProviderModel.model(jsonObject: jsonObject)
    }
    
}

class ListDataProviderModel : GLBSwiftyModel {
    
    public var uid: String = ""
    public var title: String = ""
    
    override func fromJson(json: GLBJson) {
        self.uid <<< (json, "id")
        self.title <<< (json, "title")
    }
    
}
